enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberStateMachineTest.java

```diff
@@ -463,6 +504,171 @@ public Object answer( InvocationOnMock invocation ) throws Throwable
 
     }
 
+    @Test
+    public void whenHAModeSwitcherSwitchesToSlaveTheOtherModeSwitcherDoNotGetTheOldMasterClient() throws Throwable
+    {
+        InstanceId me = new InstanceId( 1 );
+        StoreId storeId = new StoreId();
+        HighAvailabilityMemberContext context = mock( HighAvailabilityMemberContext.class );
```

Comment:
This looks like unsupported and unreadable conglomeration of mocks, can we extract some of those into test helper abstractions to make it reusable for future tests? @davidegrohmann WDYT?